pkg/blockstorage/azure/azuredisk.go

```diff
@@ -314,42 +306,38 @@ func (s *AdStorage) SnapshotCreate(ctx context.Context, volume blockstorage.Volu
 	if err != nil {
 		return nil, errors.Wrapf(err, "Could not get region from zone %s", volume.Az)
 	}
-	createSnap := azcompute.Snapshot{
-		Name:     azto.StringPtr(snapName),
-		Location: azto.StringPtr(region),
-		Tags:     *azto.StringMapPtr(tags),
-		SnapshotProperties: &azcompute.SnapshotProperties{
-			CreationData: &azcompute.CreationData{
-				CreateOption:     azcompute.Copy,
-				SourceResourceID: azto.StringPtr(volume.ID),
+	createSnap := armcompute.Snapshot{
+		Name:     blockstorage.StringPtr(snapName),
+		Location: blockstorage.StringPtr(region),
+		Tags:     *blockstorage.StringMapPtr(tags),
+		Properties: &armcompute.SnapshotProperties{
+			CreationData: &armcompute.CreationData{
+				CreateOption:     azto.Ptr(armcompute.DiskCreateOptionCopy),
+				SourceResourceID: blockstorage.StringPtr(volume.ID),
 			},
 		},
 	}
-	result, err := s.azCli.SnapshotsClient.CreateOrUpdate(ctx, s.azCli.ResourceGroup, snapName, createSnap)
+	pollerResp, err := s.azCli.SnapshotsClient.BeginCreateOrUpdate(ctx, s.azCli.ResourceGroup, snapName, createSnap, nil)
 	if err != nil {
 		return nil, errors.Wrapf(err, "Failed to create snapshot for volume %v", volume)
 	}
-	err = result.WaitForCompletionRef(ctx, s.azCli.SnapshotsClient.Client)
+	resp, err := pollerResp.PollUntilDone(ctx, nil)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Failed to create snapshot for volume %v", volume)
+		return nil, errors.Wrapf(err, "Failed to Get Snapshot after create, snaphotName %s", snapName)
 	}
-	rs, err := result.Result(*s.azCli.SnapshotsClient)
+	blockSnapshot, err := s.snapshotParse(ctx, resp.Snapshot)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Error in getting result of Snapshot create operation, snaphotName %s", snapName)
+		return nil, errors.Wrapf(err, "Failed to Parse Snapshot, snaphotName %s", snapName)
 	}
 
-	snap, err := s.SnapshotGet(ctx, azto.String(rs.ID))
-	if err != nil {
-		return nil, errors.Wrapf(err, "Failed to Get Snapshot after create, snaphotName %s", snapName)
-	}
-	snap.Volume = &volume
-	return snap, nil
+	blockSnapshot.Volume = &volume
+	return blockSnapshot, nil
 }
 
 func (s *AdStorage) SnapshotCreateWaitForCompletion(ctx context.Context, snap *blockstorage.Snapshot) error {
 	err := poll.Wait(ctx, func(ctx context.Context) (bool, error) {
 		snapshot, err := s.SnapshotGet(ctx, snap.ID)
-		if err == nil && snapshot.ProvisioningState == string(azcompute.ProvisioningStateSucceeded) {
+		if err == nil && snapshot.ProvisioningState == string(armstorage.ProvisioningStateSucceeded) {
```

Comment:
I questioned whether or not this is the correct package. When we look at the original code, the consts `azcompute.ProvisioningStateSucceeded` is part of a group of 6 const as such to indicate Provisioning states
```go
// ProvisioningState1 enumerates the values for provisioning state 1.
type ProvisioningState1 string

const (
	// ProvisioningState1Creating ...
	ProvisioningState1Creating ProvisioningState1 = "Creating"
	// ProvisioningState1Deleting ...
	ProvisioningState1Deleting ProvisioningState1 = "Deleting"
	// ProvisioningState1Failed ...
	ProvisioningState1Failed ProvisioningState1 = "Failed"
	// ProvisioningState1Migrating ...
	ProvisioningState1Migrating ProvisioningState1 = "Migrating"
	// ProvisioningState1Succeeded ...
	ProvisioningState1Succeeded ProvisioningState1 = "Succeeded"
	// ProvisioningState1Updating ...
	ProvisioningState1Updating ProvisioningState1 = "Updating"
)
```
When we look at the migrated constant `armstorage.ProvisioningStateSucceeded` here, it is part of the following group, which lacks states such as `Deleting` or `Updating`
```go
// ProvisioningState - Gets the status of the storage account at the time the operation was called.
type ProvisioningState string

const (
	ProvisioningStateCreating     ProvisioningState = "Creating"
	ProvisioningStateResolvingDNS ProvisioningState = "ResolvingDNS"
	ProvisioningStateSucceeded    ProvisioningState = "Succeeded"
)

```

Reply:
I think a more appropriate choice is [GalleryProvisioningState](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4#GalleryProvisioningState)
```go
// GalleryProvisioningState - The provisioning state, which only appears in the response.
type GalleryProvisioningState string

const (
	GalleryProvisioningStateCreating  GalleryProvisioningState = "Creating"
	GalleryProvisioningStateDeleting  GalleryProvisioningState = "Deleting"
	GalleryProvisioningStateFailed    GalleryProvisioningState = "Failed"
	GalleryProvisioningStateMigrating GalleryProvisioningState = "Migrating"
	GalleryProvisioningStateSucceeded GalleryProvisioningState = "Succeeded"
	GalleryProvisioningStateUpdating  GalleryProvisioningState = "Updating"
)
```